### PR TITLE
[WIP] unified g-buffer code for maintenance

### PR DESCRIPTION
**Pull request by lhog**
This is for review only. Not ready for merge yet.
I'll also port it to develop once I have tested the code enough with maintenance.

### CODE REVIEW

**Comment by Anarchid** [comment]
Aww yeah!!

**Comment by lhog** [comment]
The former approach was mostly ok (except for slight performance impact). The issue arose with multisampled buffers, as they are much more expensive to read from in each widget and currently not possible to write to (because lack of callouts and GLSL 4.00)

File: rts/Rendering/WorldDrawer.cpp [hunk 1]
```diff
@@ -48,6 +49,12 @@
 void CWorldDrawer::InitPre() const
 {
 	CShaderHandler::GetInstance(0);
+
+	loadscreen->SetLoadMessage("Creating Deferred Buffers");
+	GL::GeometryBufferUni::geomBuffer = new GL::GeometryBuffer("UNIFIED - GBUFFER");
+	if (GL::GeometryBufferUni::geomBuffer->EnabledAndValid(true))
```
Comment (rtri) [inline rts/Rendering/WorldDrawer.cpp]: these lines and the `SafeDelete` downstairs can go in their own functions `GL::GeometryBufferUni::Create()` and `GL::GeometryBufferUni::Destroy()`

**Comment by lhog** [comment]
Is there something else missing except for

>>> these lines and the SafeDelete downstairs can go in their own functions GL::GeometryBufferUni::Create() and GL::GeometryBufferUni::Destroy()

,which I couldn't make work oddly enough?

**Comment by lhog** [comment]
Closing due to inactivity.

**Pull request closed without merging**